massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -1105,28 +1123,46 @@ impl RetrievalThread {
                 }
                 // Add new peers
                 for peer_id in peers_connected {
-                    if !cache_write.blocks_known_by_peer.contains(&peer_id) {
+                    if cache_write.blocks_known_by_peer.get(&peer_id).is_none() {
                         //TODO: Change to detect the connection before
-                        cache_write.blocks_known_by_peer.put(
+                        cache_write.blocks_known_by_peer.insert(
                             peer_id.clone(),
                             (
-                                LruCache::new(
-                                    NonZeroUsize::new(self.config.max_node_known_blocks_size)
+                                LruMap::new(ByLength::new(
+                                    self.config
+                                        .max_node_known_blocks_size
+                                        .try_into()
                                         .expect("max_node_known_blocks_size in config must be > 0"),
-                                ),
+                                )),
                                 Instant::now(),
                             ),
                         );
                     } else {
-                        cache_write.blocks_known_by_peer.promote(&peer_id);
+                        let data = cache_write
```

Comment:
[get](https://docs.rs/schnellru/0.2.1/schnellru/struct.LruMap.html#method.get) can be used to promote also